spec/connectors/mongodb/mongo_rules_parser_spec.rb

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+# frozen_string_literal: true
+
+require 'connectors/mongodb/mongo_rules_parser'
+
+describe Connectors::MongoDB::MongoRulesParser do
+  let(:rules) { [] }
+  let(:policy) { '' }
+  let(:operator) { '' }
+
+  subject { described_class.new(rules) }
+
+  context 'with one rule' do
+    let(:rules) { [{ field: 'foo', value: 'bar', policy: policy, rule: operator }] }
+    context 'on include rule' do
+      let(:policy) { 'include' }
+      context 'equals' do
+        let(:operator) { 'Equals' }
+        it 'parses rule as equals' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => 'bar' })
+        end
+      end
+      context 'greater' do
+        let(:operator) { '>' }
+        it 'parses rule as greater' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => { '$gt' => 'bar' } })
+        end
+      end
+      context 'less' do
+        let(:operator) { '<' }
+        it 'parses rule as less' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => { '$lt' => 'bar' } })
+        end
+      end
+    end
+    context 'on exclude rule' do
+      let(:policy) { 'exclude' }
+      context 'equals' do
+        let(:operator) { 'Equals' }
+        it 'parses rule as not equals' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => { '$ne' => 'bar' } })
+        end
+      end
+      context 'greater' do
+        let(:operator) { '>' }
+        it 'parses rule as less or equals' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => { '$lte' => 'bar' } })
+        end
+      end
+      context 'less' do
+        let(:operator) { '<' }
+        it 'parses rule as less or equals' do
+          result = subject.parse
+          expect(result).to match({ 'foo' => { '$gte' => 'bar' } })
+        end
+      end
+    end
+  end
+
+  context 'with multiple rules' do
+    let(:rules) do
+      [
+        { field: 'foo', value: 'bar1', policy: 'include', rule: 'Equals' },
+        { field: 'foo', value: 'bar2', policy: 'exclude', rule: '>' }
+      ]
+    end
+    it 'parses rules as and' do
+      result = subject.parse
+      expect(result).to match({ '$and' => [{ 'foo' => 'bar1' }, { 'foo' => { '$lte' => 'bar2' } }] })
```

Comment:
This is a good example of a rule that should cause us to issue warnings and _not_ use these for pre-filtering when we get to https://github.com/elastic/enterprise-search-team/issues/3174. Because it doesn't make any sense to AND rules where the first asserts a strict value of `foo`, and the second asserts a weaker range.

For this PR, this is great through. Really nice test coverage.